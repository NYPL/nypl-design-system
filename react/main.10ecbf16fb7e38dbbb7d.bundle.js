(window.webpackJsonp=window.webpackJsonp||[]).push([[0],{1292:function(n,o,c){"use strict";c.r(o);c(1293),c(1323),c(1335),c(1442),c(1485)},1448:function(n,o){},1449:function(n,o){},1450:function(n,o){},1451:function(n,o){},501:function(n,o,c){c(502),c(645),n.exports=c(1292)},564:function(n,o){}},[[501,1,2]]]);