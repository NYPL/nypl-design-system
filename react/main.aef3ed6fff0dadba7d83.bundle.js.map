{"version":3,"file":"main.aef3ed6fff0dadba7d83.bundle.js","sources":["webpack:///main.aef3ed6fff0dadba7d83.bundle.js"],"mappings":"AAAA","sourceRoot":""}