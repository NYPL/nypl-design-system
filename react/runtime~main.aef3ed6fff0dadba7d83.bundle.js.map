{"version":3,"file":"runtime~main.aef3ed6fff0dadba7d83.bundle.js","sources":["webpack:///runtime~main.aef3ed6fff0dadba7d83.bundle.js"],"mappings":"AAAA","sourceRoot":""}