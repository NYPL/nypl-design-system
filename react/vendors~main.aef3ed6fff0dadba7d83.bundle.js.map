{"version":3,"file":"vendors~main.aef3ed6fff0dadba7d83.bundle.js","sources":["webpack:///vendors~main.aef3ed6fff0dadba7d83.bundle.js"],"mappings":";AAAA","sourceRoot":""}