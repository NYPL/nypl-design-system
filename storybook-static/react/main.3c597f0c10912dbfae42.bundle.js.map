{"version":3,"file":"main.3c597f0c10912dbfae42.bundle.js","sources":["webpack:///main.3c597f0c10912dbfae42.bundle.js"],"mappings":"AAAA","sourceRoot":""}