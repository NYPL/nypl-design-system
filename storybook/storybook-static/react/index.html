<!doctype html><html lang="en"><head><meta charset="utf-8"><title>Storybook</title><meta name="viewport" content="width=device-width,initial-scale=1"><style>html, body {
    overflow: hidden;
    height: 100%;
    width: 100%;
    margin: 0;
    padding: 0;
  }</style><script>/* globals window */
  /* eslint-disable no-underscore-dangle */
  try {
    if (window.top !== window) {
      window.__REACT_DEVTOOLS_GLOBAL_HOOK__ = window.top.__REACT_DEVTOOLS_GLOBAL_HOOK__;
    }
  } catch (e) {
    // eslint-disable-next-line no-console
    console.warn('unable to connect to top frame for connecting dev tools');
  }</script><style>#root[hidden],
      #docs-root[hidden] {
        display: none !important;
      }</style></head><body><div id="root"></div><div id="docs-root"></div><script>window['DOCS_MODE'] = false;</script><script src="runtime~main.00bd9d033d7a0b972fc9.bundle.js"></script><script src="vendors~main.d2b6aafef1a259d165c0.bundle.js"></script><script src="main.3bf71672a6dd35499004.bundle.js"></script></body></html>