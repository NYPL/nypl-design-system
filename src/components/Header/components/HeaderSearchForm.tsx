import { Box, chakra, useMultiStyleConfig } from "@chakra-ui/react";
import React, { useState } from "react";

import Button from "../../Button/Button";
import ButtonGroup from "../../ButtonGroup/ButtonGroup";
import Fieldset from "../../Fieldset/Fieldset";
import Form, { FormField, FormRow } from "../../Form/Form";
import Icon from "../../Icons/Icon";
import Radio from "../../Radio/Radio";
import RadioGroup from "../../RadioGroup/RadioGroup";
import TextInput from "../../TextInput/TextInput";

import {
  getEncoreCatalogURL,
  getNYPLSearchURL,
  getResearchCatalogURL,
} from "../utils/headerUtils";
import gaUtils, { gaConfig } from "../utils/googleAnalyticsUtils";

export interface HeaderSearchFormProps {
  isMobile?: boolean;
}

export type SearchOptionType =
  | "circulatingCatalog"
  | "researchCatalog"
  | "website";

/**
 * Displays the search form for the Header's search interface. On mobile, two
 * buttons are displayed and on desktop, two radio inputs are displayed.
 */
const HeaderSearchForm = chakra(
  ({ isMobile = false }: HeaderSearchFormProps) => {
    const defaultSearchRadioValue: SearchOptionType = "circulatingCatalog";
    const [placeholder, setPlaceholder] = useState<string>(
      "What would you like to find?"
    );
    const [searchInput, setSearchInput] = useState<string>("");
    const [searchOption, setSearchOption] = useState<SearchOptionType>(
      defaultSearchRadioValue
    );
    const [isSearchRequested, setIsSearchRequested] = useState<boolean>(false);
    const [isGAResponseReceived, setIsGAResponseReceived] =
      useState<boolean>(false);
    const styles = useMultiStyleConfig("HeaderSearchForm", { isMobile });
    // GASearchedRepo indicates which kind of search is sent.
    let gaSearchedRepo = "Unknown";

    const onSubmit = (e: any) => {
      e.preventDefault();
      const newGaConfig = { ...gaConfig };
      let gaSearchLabel;
      let requestUrl;

      // If there is a search input, make the request.
      if (searchInput) {
        if (searchOption === "circulatingCatalog") {
          gaSearchLabel = "Submit Circulating Catalog Search";
          gaSearchedRepo = "Encore";
          requestUrl = getEncoreCatalogURL(searchInput);
        }
        if (searchOption === "researchCatalog") {
          gaSearchLabel = "Submit Research Catalog Search";
          gaSearchedRepo = "Research Catalog";
          requestUrl = getResearchCatalogURL(searchInput);
        }
        if (searchOption === "website") {
          gaSearchLabel = "Submit Search";
          gaSearchedRepo = "DrupalSearch";
          requestUrl = getNYPLSearchURL(searchInput);
        }

        if (requestUrl && gaSearchLabel) {
          gaUtils.trackEvent("Search", gaSearchLabel);
          // Set a dynamic value for custom dimension2 for GA.
          newGaConfig.customDimensions.dimension2 = gaSearchedRepo;

          // There are three phases to handle the GA search event. We need to
          // prevent sending extra GA events after the search request is made.
          if (isSearchRequested && !isGAResponseReceived) {
            return false;
          }

          if (isSearchRequested && isGAResponseReceived) {
            window.location.assign(requestUrl);
            return true;
          }

          // If the search request is not made yet and the GA event hasn't been
          // sent yet, send the GA event, wait until it is received, and then
          // go to the search page.
          if (!isSearchRequested && !isGAResponseReceived) {
            setIsSearchRequested(true);
            // Send GA "Search" category and "QuerySent" action event
            // with custom dimensions for the type of search: Encore or Drupal.
            gaUtils.trackSearchQuerySend(
              searchInput,
              newGaConfig.customDimensions,
              () => {
                // Once the GA event is sent, go to the proper search page.
                setIsGAResponseReceived(true);
                window.location.assign(requestUrl);
              }
            );
          }
        }
      }
      // Otherwise, don't do anything and update the placeholder message.
      setPlaceholder("Please enter a search term.");
      return false;
    };

    return (
      <Box __css={styles}>
        <Form
          id="search-header"
          gap="grid.m"
          onSubmit={onSubmit}
          __css={styles.form}
        >
          <FormRow>
            <FormField gridColumn="1 / 3">
              <Fieldset
                id="fieldset-search"
                isLegendHidden
                legendText="Enter a keyword, then choose to search either the catalog or the website"
              >
                <TextInput
                  id="searchInput"
                  isRequired
                  labelText="Enter Search Keyword"
                  onChange={(e) => setSearchInput(e.target.value)}
                  placeholder={placeholder}
                  showRequiredLabel={false}
                  value={searchInput}
                  __css={styles.textInput}
                />
              </Fieldset>
            </FormField>
            <FormField gridColumn={{ base: "3", md: "4" }}>
              <ButtonGroup>
                <Button
                  aria-label="Search"
                  buttonType="pill"
                  id="search-btn"
                  onClick={onSubmit}
                  __css={styles.searchBtn}
                >
                  <Icon name="search" size="large" />
                </Button>
              </ButtonGroup>
            </FormField>
          </FormRow>
          <FormRow>
            <FormField>
              <RadioGroup
                defaultValue="circulatingCatalog"
                id="search-type"
                labelText="Type of search"
                layout={isMobile ? "column" : "row"}
                name="catalogWebsiteSearch"
                onChange={(val: SearchOptionType) => setSearchOption(val)}
                showLabel={false}
              >
                <Radio
                  id="circulatingCatalogSearch"
                  labelText="Search the Circulating Catalog"
                  value="circulatingCatalog"
                  __css={styles.radio}
                />
                <Radio
                  id="researchcatalogSearch"
                  labelText="Search the Research Catalog"
                  value="researchCatalog"
                  __css={styles.radio}
                />
                <Radio
                  id="websiteSearch"
                  labelText="Search NYPL.org"
                  value="website"
                  __css={styles.radio}
                />
              </RadioGroup>
            </FormField>
          </FormRow>
        </Form>
      </Box>
    );
  }
);

export default HeaderSearchForm;
