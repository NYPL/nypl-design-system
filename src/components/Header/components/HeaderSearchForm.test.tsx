import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { axe } from "jest-axe";
import * as React from "react";
import renderer from "react-test-renderer";

import HeaderSearchForm from "./HeaderSearchForm";

describe("HeaderSearchForm Accessibility", () => {
  it("passes axe accessibility test", async () => {
    const { container } = render(<HeaderSearchForm />);
    expect(await axe(container)).toHaveNoViolations();
  });

  it("passes axe accessibility test for mobile view", async () => {
    const { container } = render(<HeaderSearchForm isMobile />);
    expect(await axe(container)).toHaveNoViolations();
  });
});

describe("HeaderSearchForm", () => {
  // We want to spy on the `window.location.assign` method so we can know
  // that the correct search URL was called. Save the real
  // `window.location.assign` method.
  const realLocation = window.location;
  // We want to always return a timestamp date since that's get added
  // to the URL on every call. If we mock it, we'll always get the same
  // timestamp which makes testing easier.
  let realDate;

  beforeAll(() => {
    // Spy on the `window.location.assign` method.
    delete window.location;
    (window as any).location = { assign: jest.fn() };
    // Override the `Date` class so we always get the same
    // timestamp when `getTime` is called.
    const currentDate = new Date("2022-01-01");
    realDate = Date;
    (global as any).Date = class extends Date {
      constructor(date) {
        super(date);
        return currentDate;
      }
    };
  });

  afterAll(() => {
    // Restore the `window.location.assign` method.
    window.location = realLocation;
    // Restore the `Date` class.
    (global as any).Date = realDate;
  });

  describe("Desktop", () => {
    beforeEach(() => {
      render(<HeaderSearchForm />);
      jest.useFakeTimers();
    });

    // Running all pending timers and switching to real timers using Jest
    afterEach(() => {
      jest.runOnlyPendingTimers();
      jest.useRealTimers();
    });

    it("renders a form with an input, radio inputs, and a search button", () => {
      const form = screen.getByRole("form");
      const searchInput = screen.getByRole("textbox");
      const radioGroup = screen.getByRole("radiogroup");
      const searchBtn = screen.getByRole("button");

      expect(form).toBeInTheDocument();
      expect(searchInput).toBeInTheDocument();
      expect(radioGroup).toBeInTheDocument();
      expect(searchBtn).toBeInTheDocument();
      expect(searchBtn).toHaveAttribute("aria-label", "Search");
    });

    it("makes a request to the Encore catalog", async () => {
      const searchInput = screen.getByRole("textbox");
      const searchBtn = screen.getByRole("button");

      // The default value of the radio button is set to
      // "Search the Circulating Catalog".
      userEvent.type(searchInput, "cats");
      userEvent.click(searchBtn);

      // Fast-forward until all timers have been executed.
      // The SearchForm calls `gaUtils.trackSearchQuerySend` which
      // internally has a timer.
      jest.runAllTimers();

      // The first call to `window.location.assign` should be...
      expect(window.location.assign).toHaveBeenNthCalledWith(
        1,
        "https://browse.nypl.org/iii/encore/search/C__Scats__Orightresult__U?searched_from=header_search&timestamp=1640995200000&lang=eng"
      );
    });

    it("makes a request to the Research Catalog", async () => {
      const searchInput = screen.getByRole("textbox");
      const researchRadio = screen.getByText("Search the Research Catalog");
      const searchBtn = screen.getByRole("button");

      userEvent.type(searchInput, "cats");
      // Select the "Search the Research Catalog" radio button.
      userEvent.click(researchRadio);
      userEvent.click(searchBtn);

      // Fast-forward until all timers have been executed.
      // The SearchForm calls `gaUtils.trackSearchQuerySend` which
      // internally has a timer.
      jest.runAllTimers();

      // The second call to `window.location.assign` should be...
      expect(window.location.assign).toHaveBeenNthCalledWith(
        2,
        "https://www.nypl.org/research/research-catalog/search?q=cats&?searched_from=header_search&timestamp=1640995200000&lang=eng"
      );
    });

    it("makes a request to the web catalog", () => {
      const searchInput = screen.getByRole("textbox");
      const webRadio = screen.getByText("Search NYPL.org");
      const searchBtn = screen.getByRole("button");

      userEvent.type(searchInput, "cats");
      // Select the "Search NYPL.org" radio button.
      userEvent.click(webRadio);
      // Run the search.
      userEvent.click(searchBtn);

      // Fast-forward until all timers have been executed.
      // The SearchForm calls `gaUtils.trackSearchQuerySend` which
      // internally has a timer.
      jest.runAllTimers();

      // We mock `window.location.assign` before ALL tests and restore after
      // ALL tests. So we should have two calls to `window.location.assign`.
      // The third call to `window.location.assign` should be...
      expect(window.location.assign).toHaveBeenNthCalledWith(
        3,
        "//www.nypl.org/search/cats?searched_from=header_search&timestamp=1640995200000"
      );
    });
  });

  describe("Mobile", () => {
    beforeEach(() => {
      render(<HeaderSearchForm isMobile />);
    });

    it("renders a form with an input and three radios on mobile", () => {
      const form = screen.getByRole("form");
      const searchInput = screen.getByRole("textbox");
      const radios = screen.getAllByRole("radio");

      expect(form).toBeInTheDocument();
      expect(searchInput).toBeInTheDocument();
      expect(radios).toHaveLength(3);
      expect(
        screen.getByLabelText("Search the Circulating Catalog")
      ).toBeInTheDocument();
      expect(
        screen.getByLabelText("Search the Research Catalog")
      ).toBeInTheDocument();
      expect(screen.getByLabelText("Search NYPL.org")).toBeInTheDocument();
    });

    it("makes a request to the Encore catalog", () => {
      const searchInput = screen.getByRole("textbox");
      const circulatingCatalogRadio = screen.getAllByRole("radio")[0];

      userEvent.type(searchInput, "cats");
      // Select the Circulating Catalog
      userEvent.click(circulatingCatalogRadio);

      // Fast-forward until all timers have been executed.
      // The SearchForm calls `gaUtils.trackSearchQuerySend` which
      // internally has a timer.
      jest.runAllTimers();

      expect(window.location.assign).toHaveBeenNthCalledWith(
        1,
        "https://browse.nypl.org/iii/encore/search/C__Scats__Orightresult__U?searched_from=header_search&timestamp=1640995200000&lang=eng"
      );
    });

    it("makes a request to the Research Catalog", () => {
      const searchInput = screen.getByRole("textbox");
      const researchCatalogRadio = screen.getAllByRole("radio")[1];

      userEvent.type(searchInput, "cats");
      // Select the Research Catalog
      userEvent.click(researchCatalogRadio);

      // Fast-forward until all timers have been executed.
      // The SearchForm calls `gaUtils.trackSearchQuerySend` which
      // internally has a timer.
      jest.runAllTimers();

      expect(window.location.assign).toHaveBeenNthCalledWith(
        2,
        "https://www.nypl.org/research/research-catalog/search?q=cats&?searched_from=header_search&timestamp=1640995200000&lang=eng"
      );
    });

    it("makes a request to the web catalog", () => {
      const searchInput = screen.getByRole("textbox");
      const websiteRadio = screen.getAllByRole("radio")[2];

      userEvent.type(searchInput, "cats");
      // Select the Website
      userEvent.click(websiteRadio);

      // Fast-forward until all timers have been executed.
      // The SearchForm calls `gaUtils.trackSearchQuerySend` which
      // internally has a timer.
      jest.runAllTimers();

      expect(window.location.assign).toHaveBeenNthCalledWith(
        3,
        "//www.nypl.org/search/cats?searched_from=header_search&timestamp=1640995200000"
      );
    });
  });

  it("renders the UI snapshot correctly", () => {
    const headersearchForm = renderer.create(<HeaderSearchForm />).toJSON();
    const headersearchFormMobile = renderer
      .create(<HeaderSearchForm isMobile />)
      .toJSON();

    expect(headersearchForm).toMatchSnapshot();
    expect(headersearchFormMobile).toMatchSnapshot();
  });
});
