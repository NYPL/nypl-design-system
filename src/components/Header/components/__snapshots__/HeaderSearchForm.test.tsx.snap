// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderSearchForm renders the UI snapshot correctly 1`] = `
<div
  className="css-0"
>
  <form
    aria-label="form"
    className="css-16sv398"
    id="search-header"
    onSubmit={[Function]}
  >
    <div
      className="css-uo6vru"
      id="search-header-parent"
    >
      <div
        className="css-10aw3py"
        id="search-header-child0"
      >
        <div
          className="css-18yyq8w"
          id="search-header-child0-grandchild0"
        >
          <fieldset
            className="css-1xdhyk6"
            id="fieldset-search"
          >
            <legend>
              Enter a keyword, then choose to search either the catalog or the website
            </legend>
            <div
              className="css-1u8qly9"
              id="searchInput-wrapper"
            >
              <label
                className="css-1xdhyk6"
                htmlFor="searchInput"
                id="searchInput-label"
              >
                Enter Search Keyword
              </label>
              <input
                aria-required={true}
                autocomplete={null}
                className="chakra-input css-0"
                disabled={false}
                id="searchInput"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="What would you like to find?"
                required={true}
                step={null}
                type="text"
                value=""
              />
              <div
                aria-atomic={true}
                aria-live="polite"
                className="css-1xdhyk6"
                data-isinvalid={false}
                data-testid="helperErrorText"
                id="searchInput-helperText"
              />
            </div>
          </fieldset>
        </div>
        <div
          className="css-u8wr3o"
          id="search-header-child0-grandchild1"
        >
          <div
            className="chakra-stack css-6dfzd0"
          >
            <button
              aria-label="Search"
              className="chakra-button css-1xdhyk6"
              id="search-btn"
              onClick={[Function]}
              type="button"
            >
              <svg
                aria-hidden={true}
                className="chakra-icon css-1grhd2q"
                focusable={false}
                role="img"
                title="search icon"
                viewBox="0 0 24 24"
              >
                <g
                  stroke="currentColor"
                  strokeWidth="1.5"
                >
                  <path
                    d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
                    fill="none"
                    strokeLinecap="round"
                  />
                  <path
                    d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
                    fill="currentColor"
                    strokeLinecap="round"
                  />
                  <circle
                    cx="12"
                    cy="12"
                    fill="none"
                    r="11.25"
                    strokeMiterlimit="10"
                  />
                </g>
              </svg>
            </button>
          </div>
        </div>
      </div>
      <div
        className="css-1jiq10l"
        id="search-header-child1"
      >
        <div
          className="css-7ajbp2"
          id="search-header-child1-grandchild0"
        >
          <fieldset
            className="css-1u8qly9"
            id="radio-group-search-type"
          >
            <legend>
              Type of search
            </legend>
            <div
              className="chakra-radio-group css-0"
              role="radiogroup"
            >
              <div
                className="chakra-stack css-1objuxj"
              >
                <div
                  className="css-1xdhyk6"
                  id="circulatingCatalogSearch-wrapper"
                >
                  <label
                    className="chakra-radio css-13p0l12"
                    data-checked=""
                  >
                    <input
                      checked={true}
                      className="chakra-radio__input"
                      disabled={false}
                      id="circulatingCatalogSearch"
                      name="catalogWebsiteSearch"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      required={false}
                      style={
                        Object {
                          "border": "0px",
                          "clip": "rect(0px, 0px, 0px, 0px)",
                          "height": "1px",
                          "margin": "-1px",
                          "overflow": "hidden",
                          "padding": "0px",
                          "position": "absolute",
                          "whiteSpace": "nowrap",
                          "width": "1px",
                        }
                      }
                      type="radio"
                      value="circulatingCatalog"
                    />
                    <span
                      aria-hidden={true}
                      className="css-14wybg9"
                      data-checked=""
                      onMouseDown={[Function]}
                      onMouseEnter={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                    />
                    <span
                      className="chakra-radio__label css-1y8kf23"
                      data-checked=""
                      onMouseDown={[Function]}
                      onTouchStart={[Function]}
                    >
                      <span
                        className="css-0"
                      >
                        Search books, music, and movies
                      </span>
                    </span>
                  </label>
                  <div
                    aria-atomic={true}
                    aria-live="polite"
                    className="css-1xdhyk6"
                    data-isinvalid={false}
                    data-testid="helperErrorText"
                    id="circulatingCatalogSearch-helperText"
                  />
                </div>
                <div
                  className="css-1xdhyk6"
                  id="researchcatalogSearch-wrapper"
                >
                  <label
                    className="chakra-radio css-13p0l12"
                  >
                    <input
                      checked={false}
                      className="chakra-radio__input"
                      disabled={false}
                      id="researchcatalogSearch"
                      name="catalogWebsiteSearch"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      required={false}
                      style={
                        Object {
                          "border": "0px",
                          "clip": "rect(0px, 0px, 0px, 0px)",
                          "height": "1px",
                          "margin": "-1px",
                          "overflow": "hidden",
                          "padding": "0px",
                          "position": "absolute",
                          "whiteSpace": "nowrap",
                          "width": "1px",
                        }
                      }
                      type="radio"
                      value="researchCatalog"
                    />
                    <span
                      aria-hidden={true}
                      className="css-14wybg9"
                      onMouseDown={[Function]}
                      onMouseEnter={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                    />
                    <span
                      className="chakra-radio__label css-1y8kf23"
                      onMouseDown={[Function]}
                      onTouchStart={[Function]}
                    >
                      <span
                        className="css-0"
                      >
                        Search the Research Catalog
                      </span>
                    </span>
                  </label>
                  <div
                    aria-atomic={true}
                    aria-live="polite"
                    className="css-1xdhyk6"
                    data-isinvalid={false}
                    data-testid="helperErrorText"
                    id="researchcatalogSearch-helperText"
                  />
                </div>
                <div
                  className="css-1xdhyk6"
                  id="websiteSearch-wrapper"
                >
                  <label
                    className="chakra-radio css-13p0l12"
                  >
                    <input
                      checked={false}
                      className="chakra-radio__input"
                      disabled={false}
                      id="websiteSearch"
                      name="catalogWebsiteSearch"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      required={false}
                      style={
                        Object {
                          "border": "0px",
                          "clip": "rect(0px, 0px, 0px, 0px)",
                          "height": "1px",
                          "margin": "-1px",
                          "overflow": "hidden",
                          "padding": "0px",
                          "position": "absolute",
                          "whiteSpace": "nowrap",
                          "width": "1px",
                        }
                      }
                      type="radio"
                      value="website"
                    />
                    <span
                      aria-hidden={true}
                      className="css-14wybg9"
                      onMouseDown={[Function]}
                      onMouseEnter={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                    />
                    <span
                      className="chakra-radio__label css-1y8kf23"
                      onMouseDown={[Function]}
                      onTouchStart={[Function]}
                    >
                      <span
                        className="css-0"
                      >
                        Search the library website
                      </span>
                    </span>
                  </label>
                  <div
                    aria-atomic={true}
                    aria-live="polite"
                    className="css-1xdhyk6"
                    data-isinvalid={false}
                    data-testid="helperErrorText"
                    id="websiteSearch-helperText"
                  />
                </div>
              </div>
            </div>
            <div
              aria-atomic={true}
              aria-live="polite"
              className="css-1xdhyk6"
              data-isinvalid={false}
              data-testid="helperErrorText"
              id="search-type-helperErrorText"
            />
          </fieldset>
        </div>
      </div>
    </div>
  </form>
</div>
`;

exports[`HeaderSearchForm renders the UI snapshot correctly 2`] = `
<div
  className="css-0"
>
  <form
    aria-label="form"
    className="css-16sv398"
    id="search-header"
    onSubmit={[Function]}
  >
    <div
      className="css-uo6vru"
      id="search-header-parent"
    >
      <div
        className="css-10aw3py"
        id="search-header-child0"
      >
        <div
          className="css-18yyq8w"
          id="search-header-child0-grandchild0"
        >
          <fieldset
            className="css-1xdhyk6"
            id="fieldset-search"
          >
            <legend>
              Enter a keyword, then choose to search either the catalog or the website
            </legend>
            <div
              className="css-1u8qly9"
              id="searchInput-wrapper"
            >
              <label
                className="css-1xdhyk6"
                htmlFor="searchInput"
                id="searchInput-label"
              >
                Enter Search Keyword
              </label>
              <input
                aria-required={true}
                autocomplete={null}
                className="chakra-input css-0"
                disabled={false}
                id="searchInput"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="What would you like to find?"
                required={true}
                step={null}
                type="text"
                value=""
              />
              <div
                aria-atomic={true}
                aria-live="polite"
                className="css-1xdhyk6"
                data-isinvalid={false}
                data-testid="helperErrorText"
                id="searchInput-helperText"
              />
            </div>
          </fieldset>
        </div>
        <div
          className="css-u8wr3o"
          id="search-header-child0-grandchild1"
        >
          <div
            className="chakra-stack css-6dfzd0"
          >
            <button
              aria-label="Search"
              className="chakra-button css-1xdhyk6"
              id="search-btn"
              onClick={[Function]}
              type="button"
            >
              <svg
                aria-hidden={true}
                className="chakra-icon css-1grhd2q"
                focusable={false}
                role="img"
                title="search icon"
                viewBox="0 0 24 24"
              >
                <g
                  stroke="currentColor"
                  strokeWidth="1.5"
                >
                  <path
                    d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
                    fill="none"
                    strokeLinecap="round"
                  />
                  <path
                    d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
                    fill="currentColor"
                    strokeLinecap="round"
                  />
                  <circle
                    cx="12"
                    cy="12"
                    fill="none"
                    r="11.25"
                    strokeMiterlimit="10"
                  />
                </g>
              </svg>
            </button>
          </div>
        </div>
      </div>
      <div
        className="css-1jiq10l"
        id="search-header-child1"
      >
        <div
          className="css-7ajbp2"
          id="search-header-child1-grandchild0"
        >
          <fieldset
            className="css-1u8qly9"
            id="radio-group-search-type"
          >
            <legend>
              Type of search
            </legend>
            <div
              className="chakra-radio-group css-0"
              role="radiogroup"
            >
              <div
                className="chakra-stack css-1wdv1uh"
              >
                <div
                  className="css-1xdhyk6"
                  id="circulatingCatalogSearch-wrapper"
                >
                  <label
                    className="chakra-radio css-13p0l12"
                    data-checked=""
                  >
                    <input
                      checked={true}
                      className="chakra-radio__input"
                      disabled={false}
                      id="circulatingCatalogSearch"
                      name="catalogWebsiteSearch"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      required={false}
                      style={
                        Object {
                          "border": "0px",
                          "clip": "rect(0px, 0px, 0px, 0px)",
                          "height": "1px",
                          "margin": "-1px",
                          "overflow": "hidden",
                          "padding": "0px",
                          "position": "absolute",
                          "whiteSpace": "nowrap",
                          "width": "1px",
                        }
                      }
                      type="radio"
                      value="circulatingCatalog"
                    />
                    <span
                      aria-hidden={true}
                      className="css-14wybg9"
                      data-checked=""
                      onMouseDown={[Function]}
                      onMouseEnter={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                    />
                    <span
                      className="chakra-radio__label css-1y8kf23"
                      data-checked=""
                      onMouseDown={[Function]}
                      onTouchStart={[Function]}
                    >
                      <span
                        className="css-0"
                      >
                        Search books, music, and movies
                      </span>
                    </span>
                  </label>
                  <div
                    aria-atomic={true}
                    aria-live="polite"
                    className="css-1xdhyk6"
                    data-isinvalid={false}
                    data-testid="helperErrorText"
                    id="circulatingCatalogSearch-helperText"
                  />
                </div>
                <div
                  className="css-1xdhyk6"
                  id="researchcatalogSearch-wrapper"
                >
                  <label
                    className="chakra-radio css-13p0l12"
                  >
                    <input
                      checked={false}
                      className="chakra-radio__input"
                      disabled={false}
                      id="researchcatalogSearch"
                      name="catalogWebsiteSearch"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      required={false}
                      style={
                        Object {
                          "border": "0px",
                          "clip": "rect(0px, 0px, 0px, 0px)",
                          "height": "1px",
                          "margin": "-1px",
                          "overflow": "hidden",
                          "padding": "0px",
                          "position": "absolute",
                          "whiteSpace": "nowrap",
                          "width": "1px",
                        }
                      }
                      type="radio"
                      value="researchCatalog"
                    />
                    <span
                      aria-hidden={true}
                      className="css-14wybg9"
                      onMouseDown={[Function]}
                      onMouseEnter={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                    />
                    <span
                      className="chakra-radio__label css-1y8kf23"
                      onMouseDown={[Function]}
                      onTouchStart={[Function]}
                    >
                      <span
                        className="css-0"
                      >
                        Search the Research Catalog
                      </span>
                    </span>
                  </label>
                  <div
                    aria-atomic={true}
                    aria-live="polite"
                    className="css-1xdhyk6"
                    data-isinvalid={false}
                    data-testid="helperErrorText"
                    id="researchcatalogSearch-helperText"
                  />
                </div>
                <div
                  className="css-1xdhyk6"
                  id="websiteSearch-wrapper"
                >
                  <label
                    className="chakra-radio css-13p0l12"
                  >
                    <input
                      checked={false}
                      className="chakra-radio__input"
                      disabled={false}
                      id="websiteSearch"
                      name="catalogWebsiteSearch"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      required={false}
                      style={
                        Object {
                          "border": "0px",
                          "clip": "rect(0px, 0px, 0px, 0px)",
                          "height": "1px",
                          "margin": "-1px",
                          "overflow": "hidden",
                          "padding": "0px",
                          "position": "absolute",
                          "whiteSpace": "nowrap",
                          "width": "1px",
                        }
                      }
                      type="radio"
                      value="website"
                    />
                    <span
                      aria-hidden={true}
                      className="css-14wybg9"
                      onMouseDown={[Function]}
                      onMouseEnter={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                    />
                    <span
                      className="chakra-radio__label css-1y8kf23"
                      onMouseDown={[Function]}
                      onTouchStart={[Function]}
                    >
                      <span
                        className="css-0"
                      >
                        Search the library website
                      </span>
                    </span>
                  </label>
                  <div
                    aria-atomic={true}
                    aria-live="polite"
                    className="css-1xdhyk6"
                    data-isinvalid={false}
                    data-testid="helperErrorText"
                    id="websiteSearch-helperText"
                  />
                </div>
              </div>
            </div>
            <div
              aria-atomic={true}
              aria-live="polite"
              className="css-1xdhyk6"
              data-isinvalid={false}
              data-testid="helperErrorText"
              id="search-type-helperErrorText"
            />
          </fieldset>
        </div>
      </div>
    </div>
  </form>
</div>
`;
